tasks/post_to_database/post_to_database.py

```diff
@@ -0,0 +1,249 @@
+                               db_connect_info)
+    else:
+        raise ValueError(f"RequestMethod '{request_method}' not found.")
+
+
+def create_status_for_job_and_files(job_id: str,
+                                    granule_id: str,
+                                    request_time: str,
+                                    archive_destination: str,
+                                    files: List[Dict[str, Any]],
+                                    db_connect_info: Dict) -> None:
+    """
+    Posts the entry for the job, followed by individual entries for each file.
+
+    Args:
+        job_id: The unique identifier used for tracking requests.
+        granule_id: The id of the granule being restored.
+        archive_destination: The S3 bucket destination of where the data is archived.
+        request_time: The time the restore was requested in utc and iso-format.
+        files: A List of Dicts with the following keys:
+            'filename' (str)
+            'key_path' (str)
+            'restore_destination' (str)
+            'status_id' (int)
+            'error_message' (str, Optional)
+            'request_time' (str)
+            'last_update' (str)
+            'completion_time' (str, Optional)
+        db_connect_info: See shared_db.py's get_configuration for further details.
+    """
+    found_pending = False
+    job_completion_time = None
+    file_parameters = []
+    for file in files:
+        if file['status_id'] == OrcaStatus.PENDING.value:
+            found_pending = True
+        elif file['status_id'] == OrcaStatus.FAILED.value:
+            file_completion_time = datetime.datetime.fromisoformat(file['completion_time'])
+            if job_completion_time is None:
+                job_completion_time = file_completion_time
+            else:
+                job_completion_time = max(job_completion_time,
+                                          file_completion_time)
+        else:
+            raise ValueError(f"Status ID '{file['status_id']}' not allowed for new status.")
+        file_parameters.append({'job_id': job_id, 'granule_id': granule_id, 'filename': file['filename'],
+                                'key_path': file['key_path'], 'restore_destination': file['restore_destination'],
+                                'status_id': file['status_id'], 'error_message': file.get('error_message', None),
+                                'request_time': file['request_time'], 'last_update': file['last_update'],
+                                'completion_time': file.get('completion_time', None)})
+
+    if found_pending:
+        # Most jobs will be this. Some files are still pending.
+        job_status = OrcaStatus.PENDING
+        job_completion_time = None
+    else:
+        # All files failed during recovery request.
+        job_status = OrcaStatus.FAILED
+        job_completion_time = job_completion_time.isoformat().__str__()
+
+    engine = shared_db.get_user_connection(db_connect_info)
```

Comment:
Instead of passing `db_connect_info` throughout wondering if we shouldn't create the engine in the handler or task and then just pass the engine object to these functions `create_status_for_job_and_files` and `update_status_for_file`. Thoughts?